src/dist/replication/lib/bulk_load/replica_bulk_loader.cpp

```diff
@@ -309,28 +309,102 @@ error_code replica_bulk_loader::download_sst_files(const std::string &app_name,
 }
 
 // ThreadPool: THREAD_POOL_REPLICATION
-error_code replica_bulk_loader::parse_bulk_load_metadata(const std::string &fname,
-                                                         /*out*/ bulk_load_metadata &meta)
+error_code replica_bulk_loader::parse_bulk_load_metadata(const std::string &fname)
 {
-    // TODO(heyuchen): TBD
-    // read file and parse file content as bulk_load_metadata
+    if (!utils::filesystem::file_exists(fname)) {
+        derror_replica("file({}) doesn't exist", fname);
+        return ERR_FILE_OPERATION_FAILED;
+    }
+
+    int64_t file_sz = 0;
+    if (!utils::filesystem::file_size(fname, file_sz)) {
+        derror_replica("get file({}) size failed", fname);
+        return ERR_FILE_OPERATION_FAILED;
+    }
+
+    std::shared_ptr<char> buf = utils::make_shared_array<char>(file_sz + 1);
+    std::ifstream fin(fname, std::ifstream::in);
+    if (!fin.is_open()) {
+        derror_replica("open file({}) failed", fname);
+        return ERR_FILE_OPERATION_FAILED;
+    }
+    fin.read(buf.get(), file_sz);
+    dassert_replica(file_sz == fin.gcount(),
+                    "read file({}) failed, file_size = {} but read size = {}",
+                    fname,
+                    file_sz,
+                    fin.gcount());
+    fin.close();
+    buf.get()[fin.gcount()] = '\0';
```

Comment:
Better to make it a util in file module. See 

```cpp
error_s read_file(const std::string& path, std::string& buf)
```

Reply:
I add this function in src/core/core/filesystem.cpp